### PR TITLE
AP-2640 Employed journey feature tests

### PR DESCRIPTION
**Pull request by stephenrichards**
## What

[Link to story](https://dsdmoj.atlassian.net/browse/AP-2640)

Describe what you did and why.

## Checklist

Before you ask people to review this PR:

- Tests and rubocop should be passing: `bundle exec rake`
- Github should not be reporting conflicts; you should have recently run `git rebase main`.
- There should be no unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
- The PR description should say what you changed and why, with a link to the JIRA story.
- You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: features/providers/check_pending_employment.feature [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    Then I check "None of these"
+    Then I click 'Save and continue'
+    Then I should be on a page showing "Pending"
+    Then I click 'Save and continue'
+    And show me the page
+    Then I should be on a page showing "We need to check if John Pending can get legal aid"
+    #Then I click "Show all sections"
+    #And I should be on a page showing "Employment income"
+    #And I should be on a page showing "Fixed employment expenses deduction"
+    #And I should be on a page showing "-£45"
```
Comment (jsugarman) [inline features/providers/check_pending_employment.feature:51]: Needed? remove if not?

File: features/providers/check_pending_employment.feature [hunk 2]
```diff
@@ -0,0 +1,97 @@
+Feature: check_pending_employment
```
Comment (jsugarman) [inline features/providers/check_pending_employment.feature:1]: Consider plain english description with more context for persons less aware of business logic

File: features/providers/check_single_employment.feature [hunk 3]
```diff
@@ -0,0 +1,101 @@
+Feature: check_single_employment
```
Comment (jsugarman) [inline features/providers/check_single_employment.feature:1]: Consider plain english description with more context for persons less aware of business logic

File: spec/factories/addresses.rb [hunk 4]
```diff
@@ -5,7 +5,7 @@
     address_line_two { Faker::Address.street_address }
     city { Faker::Address.city }
     county { Faker::Address.city }
-    postcode { Faker::Address.zip }
+    postcode { ["SW10 9LB", "W6 0LQ", "SW1A 1AA", "RG2 7PU", "BH22 7HR"].sample }
```
Comment (jsugarman) [inline spec/factories/addresses.rb:8]: might be nice to configure faker for UK and use its locale specific postcode helper - have not tried this but


[Customize faker locale](https://github.com/faker-ruby/faker#customization)
Supposedly we can then [use this postcode helper](https://faker.readthedocs.io/en/master/locales/en_GB.html)

File: features/providers/check_pending_employment.feature [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+    Then I click 'Save and continue'
+    Then I should be on a page showing "Is your client prohibited from selling or borrowing against their assets?"
+    Then I choose "No"
+    Then I click 'Save and continue'
+    Then I should be on a page showing "Select if your client has received payments from these schemes or charities"
+    Then I check "None of these"
+    Then I click 'Save and continue'
+    Then I should be on a page showing "Pending"
+    Then I click 'Save and continue'
+    And show me the page
```
Comment (jsugarman) [inline features/providers/check_pending_employment.feature:46]: Needs removing.

**Review by jsugarman — COMMENTED** [review]
Looks good. 

Have  got a separate PR to merge into this to fix the failing cukes. Plus can continue to pair on this today

File: spec/factories/addresses.rb (hunk 4 above)
Comment (jsugarman) [reply spec/factories/addresses.rb:8]: Just checked this and our `Faker::Config.locale` is set to `en-GB` and so faker can generate uk postcodes using

```ruby
Faker::Address.postcode
```

File: features/providers/check_multiple_employment.feature [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+  Scenario: I am able to complete an application for an employed applicant with multiple employers
+    Given I am logged in as a provider
+    And csrf is enabled
+    And an applicant named Ida Paisley has completed his true layer interaction
+    And the system is prepped for the employed journey
+    Then I visit the applications page
+    Then I click link 'Ida Paisley'
+    Then I should be on a page showing "Continue Ida Paisley's financial assessment"
+    Then I click 'Continue'
+    Then I should be on a page showing "HMRC found a record of your client's employment"
```
[the remark this answers is not in the document]
Comment (mikekeen1) [reply features/providers/check_multiple_employment.feature]: I added line breaks and swapped out the 'Then's  for When and And

File: features/providers/check_pending_employment.feature [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+    Then I choose "Yes"
+    Then I click 'Save and continue'
+    Then I should be on a page showing "Provide details of the case"
+    Then I click 'Save and continue'
+    Then I should be on a page showing "Upload supporting evidence"
+    Then I upload an evidence file named 'hello_world.pdf'
+    Then I sleep for 2 seconds
+    Then I should be able to categorise 'hello_world.pdf' as 'Employment evidence'
+    Then I click 'Save and continue'
+    #Awaiting bug fix to allow category to not error on first click of Save and continue
```
[the remark this answers is not in the document]
Comment (mikekeen1) [reply features/providers/check_pending_employment.feature]: Swapped out for meaningful message, inc bug ticket number

File: features/providers/check_pending_employment.feature (hunk 2 above)
Comment (mikekeen1) [reply features/providers/check_pending_employment.feature]: Removed underscores and added capital C

File: features/providers/check_single_employment.feature (hunk 3 above)
Comment (mikekeen1) [reply features/providers/check_single_employment.feature]: Removed underscores and added capital C

File: features/providers/check_pending_employment.feature (hunk 1 above)
Comment (mikekeen1) [reply features/providers/check_pending_employment.feature]: Removed

File: features/providers/check_multiple_employment.feature [hunk 8]
```diff
@@ -0,0 +1,96 @@
+Feature: check_multiple_employment
```
[the remark this answers is not in the document]
Comment (mikekeen1) [reply features/providers/check_multiple_employment.feature]: Noted for next time!

File: spec/factories/addresses.rb (hunk 4 above)
Comment (mikekeen1) [reply spec/factories/addresses.rb:8]: Noted for next time

File: features/providers/check_pending_employment.feature (hunk 5 above)
Comment (mikekeen1) [reply features/providers/check_pending_employment.feature]: Removed

**Review by jsugarman — APPROVED** [review]
🎖️ Looks great IMO

There is one comment left related to using `Faker::Address.postcode` but am approving

**Pull request merged**